tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1933,6 +1933,8 @@ def stdin_publish(pub, msg_class, rate, once, filename, verbose):
     else:
         iterator = stdin_yaml_arg
 
+    number_of_messages = len(list(iterator()))
```

Comment:
For a very large file evaluating the whole iterator at the beginning (rather than incrementally) doesn't seem like a viable approach.

Reply:
Hey dirk-thomas, thanks for the feedback! I updated it by iterating up to element two (if exists)